src/app.js

```diff
@@ -221,30 +216,128 @@ export default function (app) {
 
   function patch(parent, element, oldNode, node) {
     if (oldNode == null) {
```

Comment:
Perhaps try with `if(oldNode === undefined)` (and in the following clauses) to see if that improves performance